File: src/main/java/org/zenbot/szolnok/timetable/batch/dao/BusRepository.java
package org.zenbot.szolnok.timetable.batch.dao;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenbot.szolnok.timetable.batch.domain.Bus;

public interface BusRepository extends MongoRepository<Bus, String> {

    Bus findByBusName(String busName);
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/dao/BusRepository.kt
package org.zenbot.szolnok.timetable.batch.bus.dao

import org.springframework.data.mongodb.repository.MongoRepository
import org.zenbot.szolnok.timetable.batch.bus.domain.Bus
import java.util.*

interface BusRepository : MongoRepository<Bus, String> {

    fun findByBusName(busName: String): Optional<Bus>
}
